clients/richclient/src/main/java/org/openecard/richclient/RichClient.java

```diff
@@ -492,4 +503,61 @@ private static Long getRegInt(WinReg.HKEY hk, String key, String value, @Nullabl
 	return defaultValue;
     }
 
+    private static void setJnaRuntimeDiretory() throws IOException {
+	// read value of jna.tmpdir property
+	Properties properties = new Properties(System.getProperties());
+	String propJnaTmpDir = properties.getProperty("jna.tmpdir");
+
+	// if the property has been set externally don't change it
+	if (propJnaTmpDir != null) {
+	    return;
+	}
+
+	//check if we are on Linux
+	String osName = properties.getProperty("os.name");
+	if (osName.contains("nix") || osName.contains("nux") || osName.contains("aix")) {
+
+	    // we are on linux, first read "XDG_RUNTIME_DIR" to see if a user run dir is set
+	    String userRuntimeDir = System.getenv("XDG_RUNTIME_DIR");
+	    boolean canUseNormalTempDir = true;
+	    // if it is not set, we cannot use it
+	    boolean canUseUserRunDir = (userRuntimeDir != null);
+
+	    // then read "/proc/mounts"
+	    try (BufferedReader bufferedReader = new BufferedReader(new FileReader("/proc/mounts"))) {
```

Comment:
Reading of /proc/mounts should be done seperately and the results should be saved to memory, so we can have a method that checks for the exec flag of a path. This makes it easier to change the code when needed.

Reply:
I created a new class "MountInfo" (similar to https://gist.github.com/ikonst/3394662), which can be used to parse /proc/mounts and hold the results in memory.